### PR TITLE
Pool strings

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEngineManager.cs [hunk 1]
```diff
@@ -34,6 +34,8 @@ class MSBuildEngineManager: IDisposable
 		object localLock = new object ();
 		bool disposed;
 
+		internal Core.StringInternPool Pool { get; } = new Core.StringInternPool ();
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEngineManager.cs]: This is the only shared state I can put it on, the Engine itself is null for imported projects.

File: main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs [hunk 2]
```diff
@@ -3194,7 +3194,7 @@ void ClearBracketMarkers ()
 				TextEditor.Options.ZoomChanged += value;
 			}
 			remove {
-				TextEditor.Options.ZoomChanged += value;
+				TextEditor.Options.ZoomChanged -= value;
```
Comment (KirillOsenkov) [inline main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs]: WOW. Great catch!

**Comment by slluis** [comment]
What's the impact of this change?
Also please create a build lane to verify nothing breaks before merging.

**Comment by Therzok** [comment]
https://github.com/mono/monodevelop/commit/a0b973750f1abfa03cdad0525ca8e4723907c33e

We save around 50MB of managed memory with Main.sln open.

**Comment by Therzok** [comment]
Test failures unrelated.

**Pull request merged**